lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -327,36 +342,39 @@ namespace Intl
         );
     }
 
-    HRESULT SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
+    void SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
     {
         // We know what actual type we stored in the IPlatformAgnosticResource*, so cast to it.
         auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
         icu::NumberFormat *nf = numFormatResource->GetInstance();
-        // REVIEW (doilij): Without RTTI support, we can't do dynamic_cast,
-        // which is the preferred way to know whether it was indeed created as a DecimalFormat,
-        // which is not something the API will indicate to us directly (create returns a NumberFormat which may or may not be a DecimalFormat)
-        icu::DecimalFormat *df = reinterpret_cast<icu::DecimalFormat *>(nf);
-        df->setMinimumSignificantDigits(minSigDigits);
-        df->setMaximumSignificantDigits(maxSigDigits);
-        return S_OK;
+
+        // TODO(jahorto): Determine if we could ever have a NumberFormat that isn't a DecimalFormat (and if so, what to do here)
+        icu::DecimalFormat *df = dynamic_cast<icu::DecimalFormat *>(nf);
+        if (df)
+        {
+            df->setMinimumSignificantDigits(minSigDigits);
+            df->setMaximumSignificantDigits(maxSigDigits);
+        }
+        else
+        {
+            AssertMsg(false, "Could not cast an icu::NumberFormat to an icu::DecimalFormat");
```

Comment:
Add a comment explaining why this is an Assert and not FailFast.

To wit: if we can't use DecimalFormat-specific features because we didn't get a DecimalFormat, we should not crash. Best effort is good enough for Intl outputs, and we'd assume this is a transient issue. However, non-DecimalFormat-based output might be regarded as buggy, especially if consistently wrong.  We'd like to use Debug builds to detect if this is the case and how prevalent it is, and we can make a further determination if and when we see failures here.

Reply:
That is at least as good if not better than anything I would have to say about it, so I will just copy/paste that.